crates/client/src/lib.rs

```diff
@@ -101,6 +102,44 @@ impl<R: RegistryStorage, C: ContentStorage, N: NamespaceMapStorage> Client<R, C,
         &self.namespace_map
     }
 
+    /// Get warg registry domain
+    pub async fn get_warg_registry(
+        &self,
+        namespace: &str,
+    ) -> Result<Option<RegistryDomain>, ClientError> {
+        self.update_checkpoint(None, &self.api.latest_checkpoint(None).await?, vec![])
+            .await?;
+        let operator = self
+            .registry()
+            .load_operator(Some(&RegistryDomain::from_str(namespace)?))
+            .await?;
+        if let Some(op) = operator {
+            match op.state.namespace_state(namespace) {
+                Ok(Some(warg_protocol::operator::NamespaceState::Imported { registry })) => {
+                    return Ok(Some(RegistryDomain::from_str(registry)?));
+                }
+                Ok(Some(warg_protocol::operator::NamespaceState::Defined)) => {
+                    return Ok(None);
+                }
+                Ok(None) => return Ok(None),
+                Err(e) => {
+                    return Err(ClientError::SimilarNamespace {
+                        namespace: namespace.to_string(),
+                        e: e.to_string(),
+                    });
+                }
+            }
```

Comment:
minor: its difficult to see that this `if` block always `return`s
```suggestion
            return match op.state.namespace_state(namespace) {
                Ok(Some(warg_protocol::operator::NamespaceState::Imported { registry })) => {
                    Ok(Some(RegistryDomain::from_str(registry)?))
                }
                Ok(Some(warg_protocol::operator::NamespaceState::Defined)) => {
                    Ok(None)
                }
                Ok(None) => return Ok(None),
                Err(e) => {
                    Err(ClientError::SimilarNamespace {
                        namespace: namespace.to_string(),
                        e: e.to_string(),
                    })
                }
            }
```